### PR TITLE
CORTX-34400: Execute VMWare provisioner job on docker swarm node.

### PR DESCRIPTION
**Pull request by shailesh-vaidya**
# Problem Statement
- Execute the VMWare provisioner job on the docker swarm node. 
- Added email notification

image

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- [x] Jenkins pipelines used for testing - https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/sv_space/job/provision-vm-debug/9/console

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] Jira ID/ COMMUNITY is prefixed in PR title
      Prefix Syntax - 
        `CORTX-<5 digit JIRA ID>: <PR Title>` - for Seagate Employees
        `COMMUNITY: <PR Title>`  - for open source contributors
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/automation/vra_infra_provisioner.groovy [hunk 1]
```diff
@@ -116,6 +121,19 @@ pipeline {
             script {
                 // Archive tfstate artifacts in jenkins build
                 archiveArtifacts artifacts: "solutions/vmware/terraform/*.*", onlyIfSuccessful: false, allowEmptyArchive: true
+
+                // Send email notification
+                env.vmnames = sh( script: "cat solutions/vmware/terraform/vm_details.txt", returnStdout: true).trim()
+                def toEmail = ""
```
Comment (gauravchaudhari02) [inline jenkins/automation/vra_infra_provisioner.groovy:127]: can we add a devops group mail id here?

File: jenkins/automation/vra_infra_provisioner.groovy [hunk 2]
```diff
@@ -103,8 +104,12 @@ pipeline {
                 script { build_stage = env.STAGE_NAME }
                 sh label: '', script: '''
                     pushd solutions/vmware/terraform/
+                        echo "VM list is empty" > vm_details.txt
                         terraform apply --auto-approve
-                        if [ "$?" -ne 0 ]; then echo -e '\nERROR: Infra Provision Failed!!\n'; else echo -e '\n---SUCCESS---\n'; fi
+                        if [ "$?" -ne 0 ]; then echo -e \'\\nERROR: Infra Provision Failed!!\\n\'; else echo -e \'\\n---SUCCESS---\\n\'; fi
+                        VM_HOSTNAMES=$(terraform show -json terraform.tfstate | jq .values.root_module.resources[].values.resources | jq -c \'[ .[] | select( .type | contains("Cloud.vSphere.Machine")) ]\' | tr \',\' \'\\n\' | awk -F \'\\\' \'/cloudConfig/ { print $8}\' | sed \'s/nfqdn: //g\')
```
Comment (gauravchaudhari02) [inline jenkins/automation/vra_infra_provisioner.groovy:110]: can we add comma in values of list of VM's?

Comment (shailesh-vaidya) [reply jenkins/automation/vra_infra_provisioner.groovy]: Fixed and tested with  https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/sv_space/job/provision-vm-debug/11/console

image

File: jenkins/automation/vra_infra_provisioner.groovy (hunk 1 above)
Comment (shailesh-vaidya) [reply jenkins/automation/vra_infra_provisioner.groovy]: Fixed

**Review by pankajgborole — APPROVED** [review]
Approved

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**